pkgs/data/fonts/iosevka/bin.nix

```diff
@@ -1,19 +1,27 @@
-{ stdenv, fetchzip }:
+{ stdenv, fetchurl, unzip
+, variant ? ""
+}:
 
 let
-  version = "3.4.6";
-in fetchzip {
-  name = "iosevka-bin-${version}";
+  name = "iosevka" + stdenv.lib.optionalString (variant != "") "-" + variant;
+in stdenv.mkDerivation rec {
+  pname = "${name}-bin";
+  version = "3.7.1";
 
-  url = "https://github.com/be5invis/Iosevka/releases/download/v${version}/ttc-iosevka-${version}.zip";
+  src = fetchurl {
+    url = "https://github.com/be5invis/Iosevka/releases/download/v${version}/ttc-${name}-${version}.zip";
+    sha256 = (import ./variants.nix).${name};
```

Comment:
what happens if I specify a variant that doesn't exists? should there be an `or` and a `throw` message for that?

Reply:
```
nix-repl> iosevka-bin.override { variant = "nonexistent"; }
error: attribute 'iosevka-nonexistent' missing, at /persist/home/src/nixpkgs/pkgs/data/fonts/iosevka/bin.nix:13:14
«derivation
```

I could add a throw if you'd like.

Reply:
yeah, that would be better. also, nice persistent home :grin:

Reply:
Now it looks like the following:

```
nix-repl> iosevka-bin.override { variant = "ss10"; }
«derivation /nix/store/m3ljgfq2gvsk5dvn94yxzlsyrrsx2vbk-iosevka-ss10-bin-3.7.1.drv»

nix-repl> iosevka-bin.override { variant = "nonexistent"; }
error: No such variant "nonexistent" for package iosevka-bin.
Valid variants are: aile, curly, curly-slab, etoile, slab, sparkle, ss01, ss02, ss03, ss04, ss05, ss06, ss07, ss08, ss09, ss10, ss11, ss12, ss13, ss14.

«derivation
```

ty re: home, I hadn't noticed that was visible ^_^